#!/bin/bash
#$ -P mlhd
#$ -cwd


# Snakefile
# Working dir

workdir: "../../samples"

#SAMPLES - [H_0791, H_0791]

#rule all:
 # input:
  #  expand("{sample}.sort.txt", sample = SAMPLES)

rule fastqc:
  input:
    fastq="{sample}.fastq.gz"
  output:
    zip="{sample}_fastqc.zip",
    html="{sample}_fastqc.html"
  shell:
    "fastqc {input.fastq} {output.html} {output.zip}"

rule trimmomatic:
  input:
    i1 = "{sample}.R1.fastq.gz",
    i2 = "{sample}.R2.fastq.gz"
  output:
    o1 = "{sample}.R1.trimP.fastq.gz",
    o2 = "{sample}.R1.trimU.fastq.gz",
    o3 = "{sample}.R2.trimP.fastq.gz",
    o4 = "{sample}.R2.trimU.fastq.gz",
    te = "{sample}.trimerror.txt"
  shell:
    "trimmomatic PE -phred33 {input.i1} {input.i2} {output.o1} {output.o2} {output.o3} {output.o4} ILLUMINATI:TruSeq3=PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36 2>{output.te}"

rule STAR:
    input:
      index="/projectnb/bubhub/bubhub-reference/genome/human/GENCODE/v25/GENCODE_v25_star_index",
      in1="{sample}.trimP.fastq.gz",
      in2="{sample}.trimP.fastq.gz"
    output:
      "{sample}.STAR.bam"
    shell:
      "STAR --runThreadN 8 --runMode alignReads --genomeDir {input.index} --readFilesIn {input.in1} {input.in2} --readFilesCommand zcat --outFileNamePrefix {output} --outSAMtype BAM unsorted"

#rule multiqc:
#    input:
#    output:
#   shell:
#        export LC_ALL=en_US.utf-8 export LANG=$LC_ALL""

