# Snakefile

# snakemake --cluster "qsub -P mlhd -cwd -pe omp {threads}" --jobs 3

include: '../common_workflow.snake'

bams = expand('../../samples/{sample}__trimP__STAR.Aligned.out.bam',sample=sample_names)

rule all:
  input:
    bams

#references

indexdir= '/projectnb/bubhub/bubhub-reference/genome/human/GENCODE/v25/GENCODE_v25_star_index'

rule fastqc:
  input:
    fastq="../../samples/{sample}.fastq.gz"
  output:
    zip="../../samples/{sample}__fastqc.zip",
    html="../../samples/{sample}__fastqc.html"
  shell:
    "fastqc {input.fastq} {output.html} {output.zip}"

rule trimmomatic:
  input:
    read1 = "../../samples/{sample}.R1.fastq.gz",
    read2 = "../../samples/{sample}.R2.fastq.gz",
    adapter_fa = 'TruSeq3-PE.fa'
  output:
    read1p = "../../samples/{sample}.R1__trimP.fastq.gz",
    read1u = "../../samples/{sample}.R1__trimU.fastq.gz",
    read2p = "../../samples/{sample}.R2__trimP.fastq.gz",
    read2u = "../../samples/{sample}.R2__trimU.fastq.gz"
  log:
     "../../samples/{sample}__trimerror.log"
  shell:
    "trimmomatic PE -phred33 "
    "{input.read1} {input.read2} "
    "{output.read1p} {output.read1u} {output.read2p} {output.read2u} "
    "ILLUMINACLIP:{input.adapter_fa}:2:30:10 LEADING:3 TRAILING:3 "
    "SLIDINGWINDOW:4:15 MINLEN:36 2> {log} "

rule STAR:
    input:
      index= indexdir,
      read1="../../samples/{sample}.R1__trimP.fastq.gz",
      read2="../../samples/{sample}.R2__trimP.fastq.gz"
    output:
      unsorted_bam='../../samples/{sample}__trimP__STAR.Aligned.out.bam'
    params:
      prefix='../../samples/{sample}__trimP__STAR.'
    threads: 8
    shell:
      "STAR --runThreadN {threads} "
      "--runMode alignReads "
      "--genomeDir {input.index} "
      "--readFilesIn {input.read1} {input.read2} "
      "--readFilesCommand zcat --outFileNamePrefix {params.prefix} "
      "--outSAMtype BAM Unsorted "

# rule multiqc:
#   input:
#     trim="{sample}.trimP.fastq.gz",
#     fastqc="{sample}_fastqc.zip",
#     STAR="{sample}.STAR.bam",
#     dir= multiqcdir

#   output:
#     report="{sample}.multqc_report.html"
#     prefix=/usr3/graduate/crespodi/Huntington/presymptomatic_hd_mrnaseq/samples/H_0014_BA9_mRNASeq.multiqc_report
#     samplesdir=/usr3/graduate/crespodi/Huntington/presymptomatic_hd_mrnaseq/samples/H_0014_BA9_mRNASeq*
#   shell:
#     " export LC_ALL=en_US.utf-8 export LANG=$LC_ALL "
#     " multiqc -n {output.prefix} -outdir {input.dir} {output.samplesdir} "
