* GTEX Analysis
** Information about the Data
- Genotype-Tissue Expression Project [[https://www.gtexportal.org/home/documentationPage][GTEx Website]]
- Resource to study human gene expression
- Paper [[https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4010069/][GTEx project paper]]
*Goal of experiment* 
-  Trying to create a baseline 'normal' BA9 and CAU
[[file:Brodmann_area_9.jpg][Brodmann area 9]]

[[file:CAU.png][Caudate nucleus]] 

*Donor Inclusion/Exclusion Criteria are as follows:*
1. 21 ≤ Age (years) ≤ 70
2. 18.5 < BMI < 35
3. Time between death and tissue collection less than 24 hours
4. No whole blood transfusion within 48 hours prior to death
5. No metastatic cancer
6. No chemotherapy or radiation therapy within the 2 years prior to death
7. Generally unselected for presence or absence of diseases or disorders, except for potentially communicable dis\
eases that disqualify someone to donate organs or tissues would also be disqualifying for GTEx.

*Library prep methods*
A6) Was the RNA-seq protocol for GTEx strand specific?

No. RNA-seq was performed using the Illumina TruSeq library construction protocol. 
This is a non-strand specific polyA+ selected library.  
For more details, please visit our documentation page: https://gtexportal.org/home/documentationPage


*We have a substational amount of covariate data*
[[file:phs000424.v7.pht002742.v7.p2.c1.GTEx_Subject_Phenotypes.GRU.txt][Phenotypic Data]]

#+BEGIN_SRC python :results output
import os
from xml.etree import ElementTree
import textwrap
full_file= os.path.abspath(
'phs000424.v7.pht002742.v7.GTEx_Subject_Phenotypes.data_dict.xml')

dom = ElementTree.parse(full_file)

description = dom.findall('description')
for d in description:
    long_string = d.text
for i in textwrap.wrap(long_string):
    print(i)



#+END_SRC
#+RESULTS:
#+begin_example
This subject phenotype data table includes demographic information
(n=9 variables; gender, age, race, ethnicty, height, weight, and bmi),
study eligibiliy, general medical history (n=21 variables),
autoimmune, degenerative, neurological (n=11 variables), evidence of
HIV (n=9 variables), blood donation (n=2 variables), information (n=41
variables), potential exposure: physical contact (n=3 variables),
potential exposure: sexual activity (n=4 variables), history at time
of death (n=13 variables), tissue transplant (n=2 variables), medical
history source, death circumstances (n=26 variables), tissue recovery
(n=13 variables), serology results (n=14 variables), organ donor or
post-mortem cohort, and total ischemic time for a donor.
#+end_example

** QC
*NOTE* In the trouble samples.txt there are 106 sample names, but not all samples include R1 and R2

*Analyses*

- [X] FASTQC
- [X] Trimmomatic
- [X] Salmon
- [X] STAR
- [X] Multiqc [[file:pre_filtered_multiqc_report.html][Prefiltered Multiqc Report]], [[file:Per_Base_Sequence.png][Per Base Sequence Heatmap]], [[file:/ssh:crespodi@scc4.bu.edu:/usr3/graduate/crespodi/Huntington/presymptomatic_hd_mrnaseq/analysis/GTEx/GTEX_info.tsv][GTEX Info File]], [[file:filtered_multiqc_report.html][Filtered Multiqc Report]]

#+BEGIN_SRC shell
shotwell Per_Base_Sequence.png
#+END_SRC

[[file:pre_filtered_multiqc_report.html][Prefiltered Multiqc Report]]
[[file:Per_Base_Sequence.png][Per Base Sequence Heatmap]]
- Filtered out samples due to the heatmap and being qualitatively different from the rest
- N7MT_BA9, WL46_BA9, X4EP were also filtered out which contained > 2% over represented sequences.
- T5JC_BA9, T6MN_BA9, XLM4 were filtered out because they were different lengths (250 instead of 76) 

*After filtering*
- After running it through our pipeline we went from 248 pairs to 192.
  - 51 females
  - 141 males
  - mean age = 59.4
  - median age = 60.5
  - 90 BA9 samples
  - 102 CAU samples

[[file:filtered_multiqc_report.html][Filtered Multiqc Report]]

** Differential Gene Expression
*** Previous Anaylsis 10/23/17
[[file:Snakefile][Snakefile]]

- [X] Filtered rows means less than 10 [[file:filter.py][filter.py]] [[file:GTEX_salmon_filter.csv][GTEX_salmon_filter.tsv]]
- [X] Deseq normalization  [[file:GTEX_salmon_norm.csv][GTEX_salmon_norm.csv]]
- [X] Create the design Meta data file [[file:sample_info_design.csv][sample_info_design.csv]]
- [X] Firth Logistic Regression**  [[file:OLD_GTEX_salmon_firth.tsv][GTEX_salmon_firth.tsv]], [[file:counts_beta_plot.png][Counts_Beta.png]]
- [X] David [[file:David/GOTERM_BP_DIRECT.tsv][GO_TERM_BP]], [[file:David/GOTERM_CC_DIRECT.tsv][GO_TERM_CC]], [[file:David/GOTERM_MF_FAT.tsv][GO_TERM_MF]]

*Filtering*
- After Filtering count data for rows less than mean of 10
  - Went from 58k rows down to 21,654
- Normalized using Deseq2

*Firth*

*Counts Beta*
- range between .44 and -.36
  - median of -0.000365
#+BEGIN_SRC sh
shotwell counts_beta_plot.png
#+END_SRC

#+RESULTS:

*Padjusted*
- 16,021 < .05
  - ranging .049 - 3.2e-15
  - median 3.08e-05
  - 21654/16,0021 74% 

#+RESULTS:


#+BEGIN_SRC python :results output table
import pandas as pd
import os
import textwrap
GO_BP = os.path.abspath('David/GOTERM_BP_DIRECT.tsv')

GO_BP_DF = pd.read_csv(GO_BP, sep='\t')
head = GO_BP_DF['GOTERM_BP_DIRECT'].values.tolist()
frac = head[0:20]
for idx, item in enumerate(frac):
    for j in textwrap.wrap(frac[idx]):
        print(j)
    print('\n')





#+END_SRC

*** Follow Up Analysis 

- [X] Rerun firth with more covariate data
- [X] Run the stats module on the normalized counts
- [X] Check for gender bias between brain regions
- [X] Sex stratificaiton female & male BA9 vs CAU
[[file:sex_stratification.py][sex_stratification.py]], [[file:female_sample_design_info.csv][female_sample_info]], [[file:male_sample_design_info.csv][male_sample_info]] 
[[file:stratification_anaylsis.py][stratification_analysis.py]] 

- [ ] PCA on the normalized overall matrix
- [ ] Compare old results to new
- [ ] Compare DE gene list between Asympt and the GTEx
- [ ] Fgsea?

- [ ] Recount2 get a sense of how similar are data is

*Sample Statistics*

|        | BA9 | CAU | Median Age | range |
|--------+-----+-----+------------+-------|
| Male   |  65 |  76 |         61 | 23-70 |
| Female |  25 |  26 |         60 | 44-68 |
| Total  |  90 | 102 |       60.5 |       |

*Stratificaiton Results*
| Category        | Differentially Expressed Genes |
|-----------------+--------------------------------|
| No Covariates   | 16,021                         |
| With Covariates | 15,933 (Age + Sex)             |
| Only Female     | 9981                           |
| Only Male       | 15,165                         |

*** Recount Data
[[https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1118-6][Recount Paper]]
*** About Recount
recount2 is an online resource consisting of RNA-seq gene and exon counts 
as well as coverage bigWig files for 2041 different studies. 
It is the second generation of the ReCount project. 
The raw sequencing data were processed with Rail-RNA as described in 
the recount2 paper and at Nellore et al, Genome Biology, 
2016 which created the coverage bigWig files. 
For ease of statistical analysis, for each study we created count tables at the 
gene and exon levels and extracted phenotype data, which we provide in their 
raw formats as well as in RangedSummarizedExperiment R objects 
(described in the SummarizedExperiment Bioconductor package). 
We also computed the mean coverage per study and provide it in a bigWig file, 
which can be used with the derfinder Bioconductor package to 
perform annotation-agnostic differential expression analysis 
at the expressed regions-level as described at Collado-Torres et al, 
Genome Research, 2017. The count tables, RangedSummarizeExperiment objects, 
phenotype tables, sample bigWigs, mean bigWigs, and file information tables are 
ready to use and freely available here. We also created the recount Bioconductor 
package which allows you to search and download the data for a specific study . 
By taking care of several preprocessing steps and combining many datasets into one 
easily-accessible website, we make finding and analyzing RNA-seq data considerably more 
straightforward. 

*** Notes 
- Female 2 male 1
- Library prep method poly-A
- Sex stratification, female ba9 to female cau, male ba9 to male cau.
- pull out the female brains ba9 vs caudate and then male brains
- Look at the brain regions by sex to see if there is a bias
- PCA on the overall matrix normalized 
- How samples are paired together (IE BA9 sample has a CAP sample)
- Estimate size factors Rcode

** TODO [3/3]
- [X] Remove detk from install conda and install w/ pip
- [X] Libary prep methods [[https://sites.google.com/broadinstitute.org/gtex-faqs/home][non-strand specific polyA selection]]
- [X] Housekeeping

** Hub meeting notes
11/20/17
.29 each count in the sample is divided by. 2.05 multip then twice the depth than the average
deseq2 normalization. across every gene compute geomertric mean. mult all samples together
and take the nth root.
- +change integers for sex to ['M', 'F']+
all the counts in the matrix
sequencing the brains of individuals between two population, 


11/27/17
readcount sequencing counts
- project download lots of data from different databases and process
- count for each sample and differential expression
detk-stats summary counts matrix (normalized or raw)
